extensions/common/azure/azure-blob-core/src/main/java/org/eclipse/edc/azure/blob/adapter/DefaultBlobAdapter.java

```diff
@@ -24,15 +28,22 @@
  * Implementation of {@link BlobAdapter} using a {@link BlockBlobClient}.
  */
 public class DefaultBlobAdapter implements BlobAdapter {
+    private final BlobStorageConfiguration blobStorageConfiguration;
     private final BlockBlobClient client;
 
-    public DefaultBlobAdapter(BlockBlobClient client) {
+    public DefaultBlobAdapter(BlockBlobClient client, BlobStorageConfiguration blobStorageConfiguration) {
         this.client = client;
+        this.blobStorageConfiguration = blobStorageConfiguration;
     }
 
     @Override
     public OutputStream getOutputStream() {
-        return client.getBlobOutputStream(/* overwrite = */ true);
+        var parallelTransferOptions = new ParallelTransferOptions()
+                .setBlockSizeLong(this.blobStorageConfiguration.blockSize() * Constants.MB)
+                .setMaxConcurrency(this.blobStorageConfiguration.maxConcurrency())
+                .setMaxSingleUploadSizeLong(this.blobStorageConfiguration.maxSingleUploadSize() * Constants.MB);
```

Comment:
nit: remove unnecessary `this`

Reply:
Removed in [here](https://github.com/eclipse-edc/Technology-Azure/pull/298/commits/20a8969fd1d165e7e51ecc7e2a3f2a60afc5a11a).